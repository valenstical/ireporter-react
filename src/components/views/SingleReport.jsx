/* eslint-disable jsx-a11y/media-has-caption */
import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import ago from 'date-fns/distance_in_words_strict';
import DateFormat from 'date-fns/format';


const getStatus = {
  draft: { icon: 'file-text-o', class: 'draft' },
  resolved: { icon: 'check', class: 'resolved' },
  rejected: { icon: 'remove', class: 'rejected' },
  'under investigation': { icon: 'info-circle', class: 'investigating' },
};
/**
 * The single report component
 * @returns {object} The generated JSX object
 */
export default function SingleReport({ report, handleUpdateStatus, isAdmin }) {
  const status = getStatus[report.status];
  return (
    <section className="login-wrapper top-space account-wrapper">
      <section className="heading">
        <div className="wrapper">
          <div className="bread">
            <Link to="/login">Home</Link>
            <i className="fa fa-angle-right" />
            <a className="active">{report.title}</a>
          </div>
        </div>
      </section>
      <section>
        <div className="wrapper">
          <header className="page-title" />
          <div className="reports-column">
            {isAdmin && (
            <div className="text-right" style={{ marginBottom: 20 }}>
              <select name="status" className="form-element" onChange={handleUpdateStatus} defaultValue={report.status}>
                <option value="draft" disabled>Draft</option>
                <option value="under investigation">Investigating</option>
                <option value="resolved">Resolved</option>
                <option value="rejected">Rejected</option>
              </select>
            </div>
            )}
            <div className="report-item card card-sm">
              <div className="profile-img">
                <div className="profile-img-reports" style={{ backgroundImage: `url(${report.profile})` }} />
              </div>
              <div className="report-item-content">
                <h5>{report.firstname}</h5>
                <span className="report-item-date">
                  <i className="fa fa-calendar" />
                  {' '}
                  {`${DateFormat(report.createdOn, 'MMM Do YYYY h:mm a')} (${ago(report.createdOn, new Date())} ago)`}
                </span>
                <span className={`badge-citizen badge-status status-badge-${status.class}`}>
                  <i className={`fa fa-${status.icon}`} />
                  {status.class}
                </span>
                <h6>
                  {report.title}
                  <i className={`fa flag ${report.type === 'red-flag' ? 'fa-flag flag-red-flag' : 'fa-bullhorn flag-intervention'}`} title={report.type} />
                </h6>
                <p>{report.comment}</p>
                <iframe className="map" allowFullScreen src={`https://maps.google.com/maps?q=${report.location}&output=embed&hl=en;z=20`} title={report.location} />
                {report.Videos.map(video => <video src={`${process.env.ROOT}/${video}`} controls preload="metadata" style={{ height: 364 }} key={video} />)}
                {report.Images.map(image => <img src={`${process.env.ROOT}/${image}`} alt={report.title} key={image} />)}
              </div>
            </div>
          </div>

        </div>
      </section>
    </section>


  );
}

SingleReport.propTypes = {
  report: PropTypes.object.isRequired,
  handleUpdateStatus: PropTypes.func.isRequired,
  isAdmin: PropTypes.bool
};

SingleReport.defaultProps = {
  isAdmin: false,
};
