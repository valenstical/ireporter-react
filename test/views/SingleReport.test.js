import React from 'react';
import { shallow } from 'enzyme';
import SingleReport from '../../src/components/views/SingleReport';

const props = {
  handleUpdateStatus: jest.fn(),
  details: {
    data: [
      {
        status: 'draft',
        Videos: [],
        Images: [],
      },
    ]
  }
};

describe('<SingleReport>', () => {
  it('should render without crashing', () => {
    const wrapper = shallow(<SingleReport {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render without crashing with videos and images', () => {
    props.details.data[0].Videos = [{}];
    props.details.data[0].Images = [{}];

    const wrapper = shallow(<SingleReport {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render without crashing', () => {
    const wrapper = shallow(<SingleReport {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
