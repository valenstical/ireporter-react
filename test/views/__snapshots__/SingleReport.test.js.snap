// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SingleReport> should render without crashing 1`] = `
<section
  className="login-wrapper top-space account-wrapper"
>
  <section
    className="heading"
  >
    <div
      className="wrapper"
    >
      <div
        className="bread"
      >
        <Link
          replace={false}
          to="/login"
        >
          Home
        </Link>
        <i
          className="fa fa-angle-right"
        />
        <a
          className="active"
        />
      </div>
    </div>
  </section>
  <section>
    <div
      className="wrapper"
    >
      <header
        className="page-title"
      />
      <div
        className="reports-column"
      >
        <div
          className="report-item card card-sm"
        >
          <div
            className="profile-img"
          >
            <div
              className="profile-img-reports"
              style={
                Object {
                  "backgroundImage": "url(undefined)",
                }
              }
            />
          </div>
          <div
            className="report-item-content"
          >
            <h5 />
            <span
              className="report-item-date"
            >
              <i
                className="fa fa-calendar"
              />
               
              Invalid Date (NaN years ago)
            </span>
            <span
              className="badge-citizen badge-status status-badge-draft"
            >
              <i
                className="fa fa-file-text-o"
              />
              draft
            </span>
            <h6>
              <i
                className="fa flag fa-bullhorn flag-intervention"
              />
            </h6>
            <p />
            <iframe
              allowFullScreen={true}
              className="map"
              src="https://maps.google.com/maps?q=undefined&output=embed&hl=en;z=20"
            />
          </div>
        </div>
      </div>
    </div>
  </section>
</section>
`;

exports[`<SingleReport> should render without crashing 2`] = `
<section
  className="login-wrapper top-space account-wrapper"
>
  <section
    className="heading"
  >
    <div
      className="wrapper"
    >
      <div
        className="bread"
      >
        <Link
          replace={false}
          to="/login"
        >
          Home
        </Link>
        <i
          className="fa fa-angle-right"
        />
        <a
          className="active"
        />
      </div>
    </div>
  </section>
  <section>
    <div
      className="wrapper"
    >
      <header
        className="page-title"
      />
      <div
        className="reports-column"
      >
        <div
          className="report-item card card-sm"
        >
          <div
            className="profile-img"
          >
            <div
              className="profile-img-reports"
              style={
                Object {
                  "backgroundImage": "url(undefined)",
                }
              }
            />
          </div>
          <div
            className="report-item-content"
          >
            <h5 />
            <span
              className="report-item-date"
            >
              <i
                className="fa fa-calendar"
              />
               
              Invalid Date (NaN years ago)
            </span>
            <span
              className="badge-citizen badge-status status-badge-draft"
            >
              <i
                className="fa fa-file-text-o"
              />
              draft
            </span>
            <h6>
              <i
                className="fa flag fa-bullhorn flag-intervention"
              />
            </h6>
            <p />
            <iframe
              allowFullScreen={true}
              className="map"
              src="https://maps.google.com/maps?q=undefined&output=embed&hl=en;z=20"
            />
            <video
              controls={true}
              key="[object Object]"
              preload="metadata"
              src="undefined/[object Object]"
              style={
                Object {
                  "height": 364,
                }
              }
            />
            <img
              key="[object Object]"
              src="undefined/[object Object]"
            />
          </div>
        </div>
      </div>
    </div>
  </section>
</section>
`;

exports[`<SingleReport> should render without crashing with videos and images 1`] = `
<section
  className="login-wrapper top-space account-wrapper"
>
  <section
    className="heading"
  >
    <div
      className="wrapper"
    >
      <div
        className="bread"
      >
        <Link
          replace={false}
          to="/login"
        >
          Home
        </Link>
        <i
          className="fa fa-angle-right"
        />
        <a
          className="active"
        />
      </div>
    </div>
  </section>
  <section>
    <div
      className="wrapper"
    >
      <header
        className="page-title"
      />
      <div
        className="reports-column"
      >
        <div
          className="report-item card card-sm"
        >
          <div
            className="profile-img"
          >
            <div
              className="profile-img-reports"
              style={
                Object {
                  "backgroundImage": "url(undefined)",
                }
              }
            />
          </div>
          <div
            className="report-item-content"
          >
            <h5 />
            <span
              className="report-item-date"
            >
              <i
                className="fa fa-calendar"
              />
               
              Invalid Date (NaN years ago)
            </span>
            <span
              className="badge-citizen badge-status status-badge-draft"
            >
              <i
                className="fa fa-file-text-o"
              />
              draft
            </span>
            <h6>
              <i
                className="fa flag fa-bullhorn flag-intervention"
              />
            </h6>
            <p />
            <iframe
              allowFullScreen={true}
              className="map"
              src="https://maps.google.com/maps?q=undefined&output=embed&hl=en;z=20"
            />
            <video
              controls={true}
              key="[object Object]"
              preload="metadata"
              src="undefined/[object Object]"
              style={
                Object {
                  "height": 364,
                }
              }
            />
            <img
              key="[object Object]"
              src="undefined/[object Object]"
            />
          </div>
        </div>
      </div>
    </div>
  </section>
</section>
`;
